server/game/cards/08-MotC/KakitaFavorite.js

```diff
@@ -0,0 +1,16 @@
+const DrawCard = require('../../drawcard.js');
+
+class KakitaFavorite extends DrawCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            condition: () =>
+                this.game.currentDuel &&
+                this.game.currentDuel.isInvolvedInAnyDuel(this),
```

Comment:
As a general rule, you should avoid using `this` in abilities, because it will cause problems if an effect copies this cards ability text.  `condition` gets `context`, so here what you should have is `condition: context => this.game.currentDuel && this.game.currentDuel.isInvolvedInAnyDuel(context.source)`

Reply:
got it!